tests/unit/checkpoint/test_zero_optimizer.py

```diff
@@ -336,8 +328,8 @@ def test_immediate_save_load(self, tmpdir, zero_stage):
 
     @pytest.mark.parametrize('zero_stage', [0, 1, 2, 3])
     def test_load_immediate_save(self, tmpdir, zero_stage):
-        if not get_accelerator().is_fp16_supported():
-            pytest.skip("fp16 is not supported")
+        if zero_stage == 0 and get_accelerator().device_name() == "cpu":
```

Comment:
Why special case of `zero_stage == 0`?

Reply:
this test passes on zero stage 1/2/3 but not 0.   The failure is that a view is created under no_grad then modified in place under grad mode.  Some investigation will be needed.